<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="UTF-8">
    <title>DISTIQUE by esayyari</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" type="text/css" href="stylesheets/normalize.css" media="screen">
    <link href='https://fonts.googleapis.com/css?family=Open+Sans:400,700' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/github-light.css" media="screen">
  </head>
  <body>
    <section class="page-header">
      <h1 class="project-name">DISTIQUE</h1>
      <h2 class="project-tagline">DISTIQUE by Erfan Sayyari</h2>
    </section>

    <section class="main-content">
<h3>
<a id="distique" class="anchor" href="#distique" aria-hidden="true"><span class="octicon octicon-link"></span></a>DISTIQUE</h3>
<p>DISTIQUE is a python based tool for inferring species trees from gene trees. It is a forced acronym for Distance-based Inference of Species Trees using Induced QUartet Elements.</p>
<p>The basic idea behind DISTIQUE is to infer the species tree from gene trees by computing a distance between two species from the set of quartets that include the two species. We have found ways to do this in a statistically consistent manner. DISTIQUE introduces a family of distance-based tree inference methods, with running times ranging between quadratic to quartic in the number of leaves</p>
<h3 id="how-distique-infers-species-trees"><span id="how-distique-infers-species-trees"><a href="#how-distique-infers-species-trees"><span class="octicon octicon-link"></span></a></span>How DISTIQUE infers species trees</h3>
<p>It first finds a distance matrix based on the quartet trees induced by gene trees, and then infers the species trees using a distance-based method like FASTME or PhyD. *</p>
<h3 id="how-to-install-distique"><span id="how-to-install-distique"><a href="#how-to-install-distique"><span class="octicon octicon-link"></span></a></span>How to install DISTIQUE</h3>
<p>You could install DISTIQUE in a couple of steps:<br />
 1. Clone to <a href="https://github.com/esayyari/DISTIQUE">DISTIQUE git repository</a> or download <a href="https://github.com/esayyari/DISTIQUE/archive/master.zip">this zip file</a>.<br />
 2. Then you need to set environmental variable WS_HOME to the directory under which this &quot;DISTIQUE&quot; repository is placed. There are a couple of dependencies that you need to install at this step as well.</p>
<h3 id="dependencies"><span id="dependencies"><a href="#dependencies"><span class="octicon octicon-link"></span></a></span>Dependencies</h3>
<ol style="list-style-type: decimal">
<li>FASTME 2.0 is now available at <a href="http://www.atgc-montpellier.fr/fastme/binaries.php">this link</a>. <a href="http://www.atgc-montpellier.fr/fastme/binaries.php">FASTME</a> is a distance based method that produces species trees based on the provided distance matrix. Put it under your WS_HOME.</li>
<li>PhyD* is another distance based method that could handle missing values in distance matrix reasonably. You could download it from <a href="http://www.atgc-montpellier.fr/phyd/binaries.php">this link</a>. Then put it under you WS_HOME.<br />
</li>
<li>Quartet code from <a href="http://jensjohansen.com/thesis/">this thesis</a>. Make sure that you set the QUARTETS flag to &quot;1&quot; for compiling it. After compilation put the executable code quart_bin under DISTIQUE/bin.<br />
</li>
<li>The set of tools at <a href="https://github.com/smirarab/global">this git repository</a>. They are written in different languages, but installing it is supper easy.<br />
</li>
<li>DendroPy 4.0.3, which is a Python library for phylogenetic computing.</li>
</ol>
<h3 id="how-distique-works"><span id="how-distique-works"><a href="#how-distique-works"><span class="octicon octicon-link"></span></a></span>How DISTIQUE works</h3>
<p>The main file to use DISTIQUE is available under DISTIQUE/src/utils/distique-2.py</p>
<p>Usage: python distique-2.py [-h (show help)] [-f FILENAME (pre-computed quartet table)] [-g GT (read gene trees from GT)] [-o OUT (working directory)] [-t THR (min majority consensus threshold, default is 0.5)] [-m METHOD (The method to compute the distance of taxa. The default is prod)] [-a AV (The average method to find the average quartet table. Default is mean.)] [ -v VERBOSE]</p>
<p>The Distance-Sum version of DISTIQUE is available under DISTIQUE/src/utils/distique-v5.py</p>
<p>Usage: python distique-v5.py [-h (show help)] [-f FILENAME (pre-computed quartet table)] [-g GT (read gene trees from GT)] [-o OUT (working directory)] [-t THR (min majority consensus threshold, default is 0.5)] [-m METHOD (The method to compute the distance of taxa. The default is prod)] [-a AV (The average method to find the average quartet table. Default is mean.)] [ -v VERBOSE] [-n NUM, --numStep=NUM (#rounds of anchoring, default is 2)] [-u SUMPROG (The summerize method program to build species tree from distance matrix. Default is fastme)] [-z SUMPROGOPTION (The distance method to build the tree.)]</p>
<p>The Tree-Sum version of DISTIQUE is available under DISTIQUE/src/utils/distique-v8.py</p>
<p>Usage: python distique-v8.py [-h (show help)] [-f FILENAME (pre-computed quartet table)] [-g GT (read gene trees from GT)] [-o OUT (working directory)] [-t THR (min majority consensus threshold, default is 0.5)] [-m METHOD (The method to compute the distance of taxa. The default is prod)] [-a AV (The average method to find the average quartet table. Default is mean.)] [ -v VERBOSE] [-n NUM, --numStep=NUM (#rounds of anchoring, default is 2)] [-u SUMPROG (The summerize method program to build species tree from distance matrix. Default is fastme)] [-z SUMPROGOPTION (The distance method to build the tree.)]</p>
<h3 id="datasets"><span id="datasets"><a href="#datasets"><span class="octicon octicon-link"></span></a></span>Datasets</h3>
<p>There are mainly three datasets that we used in this paper.<br />
 1. <a href="">avian dataset</a>. This dataset contains estimated gene trees, true gene trees, and true species tree<br />
 2. <a href="">mammalian dataset</a>. This dataset contains true gene trees, estimated gene trees and true species tree</p>
<h3 id="biological-dataset"><span id="biological-dataset"><a href="#biological-dataset"><span class="octicon octicon-link"></span></a></span>Biological dataset</h3>
<p>We reanalyze the avian biological dataset of Jarvis et. al. available at paper:<br />
 Jarvis, Erich D., et. al. &quot;Phylogenomic analyses data of the avian phylogenomics project.&quot; GigaScience 4.1 (2015): 1-9.</p>
<p>The resulting species trees are available at <a href="">biological species trees</a>. ### <span id="spieces-trees"><a href="#spieces-trees"><span class="octicon octicon-link"></span></a></span>Spieces trees</p>
<p>The species trees inferred using the following methods are provided.<br />
 1. <a href="">DISTIQUE-AAD</a><br />
 2. <a href="">DISTIQUE-AMD</a><br />
 2. <a href="">ASTRID (NJst)</a><br />
 3. <a href="">ASTRAL</a><br />
 ### <span id="report-bugs"><a href="#report-bugs"><span class="octicon octicon-link"></span></a></span>Report bugs</p>
<p>contact <span class="citation">[@esayyari]</span>(https://github.com/esayyari) (<script type="text/javascript">
<!--
h='&#x65;&#110;&#x67;&#46;&#x75;&#x63;&#x73;&#100;&#46;&#x65;&#100;&#x75;';a='&#64;';n='&#x65;&#x73;&#x61;&#x79;&#x79;&#x61;&#114;&#x69;';e=n+a+h;
document.write('<a h'+'ref'+'="ma'+'ilto'+':'+e+'" clas'+'s="em' + 'ail">'+e+'<\/'+'a'+'>');
// -->
</script><noscript>&#x65;&#x73;&#x61;&#x79;&#x79;&#x61;&#114;&#x69;&#32;&#x61;&#116;&#32;&#x65;&#110;&#x67;&#32;&#100;&#x6f;&#116;&#32;&#x75;&#x63;&#x73;&#100;&#32;&#100;&#x6f;&#116;&#32;&#x65;&#100;&#x75;</noscript>).</p>
<p><span class="site-footer-credits">This page was generated by <a href="https://pages.github.com">GitHub Pages</a> using the <a href="https://github.com/jasonlong/cayman-theme">Cayman theme</a> by <a href="https://twitter.com/jasonlong">Jason Long</a>.</span></p>
</div>
